pyvista/examples/downloads.py

```diff
@@ -10,304 +10,364 @@
+        'mars.jpg': None,
+        'masonry.bmp': None,
+        'mesh_fs8.exo': None,
+        'model_with_variance.vtu': None,
+        'moonlanding.png': None,
+        'motor.g': None,
+        'mug.e': None,
+        'multi-bin.f': None,
+        'multi-bin.q': None,
+        'multi-bin.xyz': None,
+        'multi.p3d': None,
+        'nefertiti.ply.zip': None,
+        'notch_disp.vtu': None,
+        'notch_stress.vtk': None,
+        'nut.slc': None,
+        'office.binary.vtk': None,
+        'osmnx_graph.p': None,
+        'parched_canal_4k.hdr': None,
+        'pepper.ply': None,
+        'pine_root.tri': None,
+        'polyline.vtk': None,
+        'probes.vtp': None,
+        'prostate.img': None,
+        'puppy.jpg': None,
+        'puppy.jpg': None,
+        'pvtu_blood_vessels/blood_vessels.zip': None,
+        'room_surface_mesh.obj': None,
+        'shark.ply': None,
+        'skybox-nz.jpg': None,
+        'skybox-nz.jpg': None,
+        'skybox2-negx': None,
+        'skybox2-negx.jpg': None,
+        'skybox2-negy': None,
+        'skybox2-negy.jpg': None,
+        'skybox2-negz': None,
+        'skybox2-negz.jpg': None,
+        'skybox2-posx': None,
+        'skybox2-posx.jpg': None,
+        'skybox2-posy': None,
+        'skybox2-posy.jpg': None,
+        'skybox2-posz': None,
+        'skybox2-posz.jpg': None,
+        'sparsePoints.txt': None,
+        'spider.ply': None,
+        'stars.jpg': None,
+        'teapot.g': None,
+        'tecplot_ascii.dat': None,
+        'tensors.vtk': None,
+        'tigerfighter.obj': None,
+        'trumpet.obj': None,
+        'turbineblade.ply': None,
+        'uGridEx.vtk': None,
+        'urn.stl': None,
+        'usa.vtk': None,
+        'usa_image.jpg': None,
+        'vrml/sextant.wrl': None,
+        'vrml/teapot.wrl': None,
+        'vtk.png': None,
+        'vtk.vtp': None,
+        'vw_knee.slc': None,
+        'woman.stl': None,
+    },
+    retry_if_failed=3,
+)
+
+
+def file_from_files(target_file, fnames):
+    """Return the full path of a single file within a list of files.
+
+    Parameters
+    ----------
+    target_file : str
+        File base name to find.
+
+    fnames : list
+        List of filenames.
 
     Returns
     -------
-    bool
-        Returns ``True``.
-
-    Examples
-    --------
-    Delete all local downloads.
-
-    >>> from pyvista import examples
-    >>> examples.delete_downloads()  # doctest:+SKIP
-    True
+    str
+        Entry in ``fnames`` matching ``filename``.
 
     """
-    _check_examples_path()
-    if os.path.isdir(pyvista.EXAMPLES_PATH):
-        shutil.rmtree(pyvista.EXAMPLES_PATH)
-    os.makedirs(pyvista.EXAMPLES_PATH)
-    return True
-
-
-def _decompress(filename, output_path):
-    _check_examples_path()
-    zip_ref = zipfile.ZipFile(filename, 'r')
-    zip_ref.extractall(output_path)
-    return zip_ref.close()
+    for fname in fnames:
+        if os.path.basename(fname) == target_file:
```

Comment:
The OpenFoam tests are failing because there are multiple matches here in a zip file.  We need to match the full path.

Reply:
Could we use `members` of the pooch's `Unzip` class to work around this? https://www.fatiando.org/pooch/latest/api/generated/pooch.Unzip.html

Reply:
We need >30 files with preserved folder structure for the OpenFoam tests.  While it is possible to work around it this way, it would require specifying all the files.  IMO, it would be more streamlined to optionally match a full or partial path for the one file that is needed to specify for the reader, but I'm not insisting on it.

If we do keep the current functionality of matching any `basename` or a partial path, then we should raise or warn if multiple files would match.  As we see, this will cause different results on different OS.

Reply:
The most important thing is that we will want to use the other data in this zip file when new functionality is added to `OpenFoamReader`.  When this happens, we wouldn't be able to exclude anything using `members`.